Biasing/src/Biasing/TaggerVetoFilter.cxx

```diff
@@ -13,11 +13,20 @@ TaggerVetoFilter::TaggerVetoFilter(const std::string& name,
                                    framework::config::Parameters& parameters)
     : simcore::UserAction(name, parameters) {
   threshold_ = parameters.getParameter<double>("threshold");
+  reject_primaries_missing_tagger_ =
+      parameters.getParameter<bool>("reject_events_missing_tagger");
 }
 
 TaggerVetoFilter::~TaggerVetoFilter() {}
 
-void TaggerVetoFilter::stepping(const G4Step* step) {
+void TaggerVetoFilter::EndOfEventAction(const G4Event *) {
+  if (reject_primaries_missing_tagger_) {
+    if (!getEventInfo()->primaryEnteredTaggerRegion()) {
+      G4RunManager::GetRunManager()->AbortEvent();
+    }
```

Comment:
Is there a reason to have the `primaryEnteredTaggerRegion` flag be in the event info and not just a part of the `TaggerVetoFilter` class?

All `EVENT` actions come with `Begin` and `End` of event actions so you could reset the flag in `Begin` and check if the event should be aborted in `End`.

Reply:
The reason for that would be that I made the addition to event info before realizing that you could have an action be multiple types :) 

Doing it with the begin action would have the benefit of only touching one module.

Reply:
I would switch to having `TaggerVetoFilter` holding the flag unless you can think of a reason for this flag to be used by other UserActions or be stored in the output (those are the two reasons I think it is useful to put something in EventInfo).

Reply:
Should be there now!